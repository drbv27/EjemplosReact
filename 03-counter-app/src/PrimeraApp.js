import React from "react";

//Functional Components

const PrimeraApp = () => {
  return (
    <>
      <h1>Hola Mundo</h1>
      <p>Mi primera aplicacion</p>
    </>
  );
};

export default PrimeraApp;
