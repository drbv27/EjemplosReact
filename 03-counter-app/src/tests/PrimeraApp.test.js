//import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

describe("Pruebas en <PrimeraApp/>", () => {
  // test("Debe de mostrar el mensaje Hola, soy Gokú", () => {
  //   const saludo = "Hola, soy Gokú";
  //   //const wrapper = render(<PrimeraApp saludo={saludo} />);
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });
  test("Debe de mostrar <PrimeraApp/> corectamente", () => {
    const saludo = "Hola, soy Gokú";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });
  test("debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola, soy Gokú";
    const subTitulo = "Soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );
    const textoParrafo = wrapper.find("p").text();
    //console.log(textoParrafo);

    expect(textoParrafo).toBe(subTitulo);
  });
});
