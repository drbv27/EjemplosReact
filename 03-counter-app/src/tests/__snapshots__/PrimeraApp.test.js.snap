// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de mostrar <PrimeraApp/> corectamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Gok√∫
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
